### DISCUSSION TITLE
Cannot implicitly convert type 'Melanchall.DryWetMidi.Interaction.ITimeSpan' to 'Melanchall.DryWetMidi.Interaction.MetricTimeSpan'

### DISCUSSION THREAD

**Discussion by TravisWoolston** [melanchall/drywetmidi#290] [Q&A] [answered]
Hello, I'm new to DryWetMidi and I'm working on getting my note grid set up for my game. I'd like to create a script to render notes like guitar hero to start. I believe TrackNotes might be a better solution to this, if you can direct me to an example of that utilization. Any way, I'm taking a beginner approach and trying to subtract 2 seconds of MetricTimeSpan from a note to render it on my guitar hero track.

Docs: https://melanchall.github.io/drywetmidi/articles/high-level-managing/Time-and-length.html#metric

Script: 

public class MidiGameManager : MonoBehaviour
{
    public string midiFilePath;
    public GameObject circlePrefab; // Assign a prefab representing the circle in the Inspector
    public Transform linePosition; // Assign the position of the horizontal line in the Inspector
    public float tempo;

    private MidiFile midiFile;
    private IEnumerable<Note> notes;
    private TempoMap tempoMap;

    void Start()
    {
        // Load MIDI file
        midiFile = MidiFile.Read(midiFilePath);

        // Create a tempo map with the specified tempo
        tempoMap = TempoMap.Create(Tempo.FromBeatsPerMinute(tempo));

        // Parse MIDI tracks and extract notes
        notes = midiFile.GetNotes();

        // Start gameplay
        StartCoroutine(PlayNotes());
    }

    IEnumerator PlayNotes()
    {
        foreach (var note in notes)
        {
            // Calculate timing based on note's time and tempo
            MetricTimeSpan noteTiming = note.TimeAs<MetricTimeSpan>(tempoMap);

            // Convert 2 seconds to MetricTimeSpan
            MetricTimeSpan circleOffset = new MetricTimeSpan(0, 0, 2);

            // Calculate timing for circle instantiation (2 seconds before the note)
            MetricTimeSpan circleTiming = noteTiming.Subtract(circleOffset, TimeSpanMode.TimeTime);

            // Wait for the circle to be created
            yield return new WaitForSeconds((float)circleTiming.TotalMicroseconds / 1000000f);

            // Create a circle GameObject at the line position
            GameObject circle = Instantiate(circlePrefab, linePosition.position, Quaternion.identity);

            // Position the circle according to the MIDI note timing
            Vector3 newPosition = circle.transform.position;
            newPosition.x = (float)noteTiming.TotalMicroseconds / 1000000f; // Adjust x position based on note timing
            circle.transform.position = newPosition;
        }
    }
}

**Comment by melanchall** [comment, marked as answer]
Hi,

You should use `TimeSpanMode.TimeLength`.
